ggshield/cmd/honeytoken/create_with_context.py

```diff
@@ -0,0 +1,147 @@
+        if path.is_dir() and depth < max_depth:
+            ext_list.extend(_list_extensions_in_dir(path, max_depth, depth + 1))
+    return ext_list
+
+
+def _get_most_common_extensions_in_dir(dir: Path):
+    """
+    Return the most 3 common extensions in a directory
+    """
+    counter = Counter(_list_extensions_in_dir(dir))
+    return [extension for extension, _ in counter.most_common(3)]
+
+
+@click.command()
+@click.option(
+    "--name",
+    "name",
+    required=False,
+    type=str,
+    help="Specify a name for your honeytoken. If this option is not provided, a unique name will be generated with a \
+`ggshield-` prefix.",
+)
+@click.option(
+    "--type",
+    "type_",
+    required=True,
+    type=click.Choice(("AWS",)),
+    help="Specify the type of honeytoken that you want to create. (For now only AWS honeytokens are supported)",
+)
+@click.option(
+    "--description",
+    "description",
+    required=False,
+    type=str,
+    help="Add a description to your honeytoken (250 characters max).",
+)
+@click.option(
+    "--language",
+    "language",
+    required=False,
+    type=str,
+    help="Language to use for the context. If not set, ggshield infers the language \
+        from the repository or from OUTPUT_FILE, if set.",
+)
+@click.option(
+    "-o",
+    "--output",
+    "output_file",
+    type=RealPath(file_okay=True, dir_okay=False, readable=True, writable=True),
+    required=False,
+    help="Filename to store your honeytoken.",
+)
+@add_common_options()
+@click.pass_context
+def create_with_context_cmd(
+    ctx: click.Context,
+    name: Optional[str],
+    type_: str,
+    description: Optional[str],
+    output_file: Optional[Path],
+    language: Optional[str],
+    **kwargs: Any,
+) -> int:
+    """
+    Create a honeytoken within a context.
+
+    The context is realistic file in which your honeytoken is inserted.
+    Adding your honeytoken within a relevant context makes it look more credible.
+
+    The prerequisites to use this command are the following:
+
+    - you have the necessary permissions as a user (for now, Honeytoken is
+    restricted to users with a manager role),
+
+    - the personal access token used by ggshield has the `honeytokens:write`
+    scope.
+    """
+    # if name is not given, generate a random one
+    if not name:
+        name = generate_random_honeytoken_name()
+    ctx_obj = ContextObj.get(ctx)
+    client = create_client_from_config(ctx_obj.config, ctx_obj.ui)
+    common_options = {"name": name, "type_": type_, "description": description}
+
+    if language or output_file:
+        options = {
+            **common_options,
+            "language": language,
+            "filename": output_file.name if output_file else None,
+        }
+    else:
+        options = {
+            **common_options,
+            "project_extensions": _get_most_common_extensions_in_dir(Path(".")),
+        }
+
+    response = client.create_honeytoken_with_context(**options)
```

Comment:
I would pass common options straight to the function, this way the type checker can help you a bit more, as it is able to verify at least common options exist, and are of the right type.

```suggestion
    if language or output_file:
        options = {
            "language": language,
            "filename": output_file.name if output_file else None,
        }
    else:
        options = {
            "project_extensions": _get_most_common_extensions_in_dir(Path(".")),
        }

    response = client.create_honeytoken_with_context(
        name=name,
        type_=type_,
        description=description,
        **options,
    )
```

Reply:
I managed to remove kvargs for better type checking.